### PR TITLE
Add user profile page

### LINKED ISSUE #1109: User public profile page
### User Profile Page

User profile page should display the basic info of the user and the courses related to the user, and active courses and completed courses should be displayed separately ( See #623 ).

Since we have multiple instance, information like **courses should be scoped to current instance**. We should not display the course information from other instances.

And, if a global admin visits the page, instances related to this user should also be displayed.

### LINKED ISSUE #2067: Course owner not linked properly in instance admin page
In the `<URL>/admin/instance/courses` page, the listed course owners do not link to their course_user profile. To fix this.

### PR DESCRIPTION
**Pull request by jeremyyap**
Fixes #2067
Closes #1109

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  private
+
+  def unscope_resources
+    Course.unscoped do
+      yield
+    end
+  end
+
+  def other_instances
+    @user.course_users.map(&:course).map(&:instance).uniq - [current_tenant]
```
Comment (allenwq) [inline app/controllers/users_controller.rb]: You can get the instances through: `ActsAsTenant.without_tenant { @user.instance_users.map(&:instance) }`, I think we should show all instances instead.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @course_users = @user.course_users.with_course_statistics
+    @current_course_users = @course_users.from_instance(current_tenant).merge(Course.current)
+    @completed_course_users = @course_users.from_instance(current_tenant).merge(Course.completed)
+    @instances = other_instances
+  end
+
+  private
+
+  def unscope_resources
+    Course.unscoped do
```
Comment (allenwq) [inline app/controllers/users_controller.rb]: better use: 
```
ActsAsTenant.without_tenant do
end
```

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+class UsersController < ApplicationController
+  around_action :unscope_resources
+  load_resource :user
+
+  def show
+    @course_users = @user.course_users.with_course_statistics
+    @current_course_users = @course_users.from_instance(current_tenant).merge(Course.current)
```
Comment (allenwq) [inline app/controllers/users_controller.rb]: Only the course_users in current tenant should be displayed to normal users, all course_users should be displayed to admins.

Also need to eager load courses associated with the user.

**Comment by weiqingtoh** [comment]
Updated 👍

**Pull request merged**